lib/kafka/consumer.rb

```diff
@@ -280,6 +280,21 @@ def each_batch(min_bytes: 1, max_wait_time: 5, automatically_mark_as_processed:
       end
     end
 
+    # @param min_bytes [Integer] the minimum number of bytes to read before
+    #   returning messages from the server; if `max_wait_time` is reached, this
+    #   is ignored.
+    # @param max_wait_time [Integer, Float] the maximum duration of time to wait before
+    #   returning messages from the server, in seconds.
+    # @param topics_offsets [Hash] the offsets to fetch from. By default it will
+    #   fetch from where the offset manager defaults too.
+    # @return batch [Kafka::FetchedBatch] a message batch fetched from Kafka.
+    def poll(min_bytes: 1, max_wait_time: 5, offsets: {})
```

Comment:
Consumer will do something like:
```
consumer.seek_to(desired_offset)
fetched_batches = consumer.poll 
```

Then i can remove this `offsets` idea from poll. Now that i think more about this, it feels cleaner than giving the responsibility of find out where to fetch from to `#poll`.